### PR TITLE
Explain consequences of using Route.seal better in scaladoc

### PR DESCRIPTION
**Pull request by jrudolph**
Replaces #766.

### CODE REVIEW

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

File: akka-http/src/main/scala/akka/http/javadsl/server/Route.scala [hunk 1]
```diff
@@ -40,12 +40,26 @@ trait Route {
   def flow(system: ActorSystem, materializer: Materializer): Flow[HttpRequest, HttpResponse, NotUsed]
 
   /**
-   * "Seals" a route by wrapping it with default exception handling and rejection conversion.
+   * Seals a route by wrapping it with default exception handling and rejection conversion.
+   *
+   * A sealed route has these properties:
+   *  - The result of the route will always be a complete response, i.e. the result of the future is a
+   *    [Success(RouteResult.Complete(response))], never a failed future and never a rejected route. These
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/javadsl/server/Route.scala]: does `[` work here? Shouldnt it be just code snippet? ````

File: akka-http/src/main/scala/akka/http/javadsl/server/Route.scala [hunk 2]
```diff
@@ -40,12 +40,26 @@ trait Route {
...
+   *    will be already be handled using the implicitly given RejectionHandler and ExceptionHandler (or
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/javadsl/server/Route.scala]: `[[RejectionHandler]]` and `[[ExceptionHandler]]`

File: akka-http/src/main/scala/akka/http/javadsl/server/Route.scala (hunk 1 above)
Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/javadsl/server/Route.scala]: Ah, stupid, I thought I was writing javadoc.

**Pull request merged**